src/app/components/login/UserPasswordReset.js

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import { graphql, commitMutation } from 'react-relay/compat';
+import PropTypes from 'prop-types';
+import { browserHistory } from 'react-router';
+import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';
+import Relay from 'react-relay/classic';
+import cx from 'classnames/bind';
+import ButtonMain from '../cds/buttons-checkboxes-chips/ButtonMain';
+import TextField from '../cds/inputs/TextField';
+import PageTitle from '../PageTitle';
+import { FormattedGlobalMessage } from '../MappedMessage';
+import CheckAgreeTerms from '../CheckAgreeTerms';
+import GenericUnknownErrorMessage from '../GenericUnknownErrorMessage';
+import ErrorBoundary from '../error/ErrorBoundary';
+import { stringHelper } from '../../customHelpers';
+import { getErrorMessageForRelayModernProblem } from '../../helpers';
+import styles from './login.module.css';
+import inputStyles from '../../styles/css/inputs.module.css';
+
+const UserPasswordReset = (props) => {
+  const [showConfirmDialog, setShowConfirmDialog] = React.useState(false);
+  const [submitDisabled, setSubmitDisabled] = React.useState(true);
+  const [errorMsg, setErrorMsg] = React.useState(null);
+  const [email, setEmail] = React.useState(null);
+
+  const handleGoBack = () => {
+    browserHistory.goBack();
+  };
+
+  const handleSignIn = () => {
+    browserHistory.push('/');
+  };
+
+  const handleChange = (e) => {
+    const value = e.target.value.trim();
+    const canSubmit = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(value);
+    const message = canSubmit ? '' : (
+      <FormattedMessage
+        id="passwordReset.emailNotValid"
+        defaultMessage="Please enter a valid email address."
+        description="Error message for invalid email address"
+      />
+    );
+    setErrorMsg(message);
+    setEmail(value);
+    setSubmitDisabled(!canSubmit);
+  };
+
+  const handleSubmit = (e) => {
+    const onFailure = (error) => {
+      const message = getErrorMessageForRelayModernProblem(error) || <GenericUnknownErrorMessage />;
+      setErrorMsg(message);
```

Comment:
This is the opposite pattern from the one I commented on above -- do we need to grab `[0].message` or just `message` here? Is there a reason it would be one way here versus the other?

Reply:
I didn't catch this, because this page doesn't use the same error display concept. most likely because of the development done at different times. To keep things simple, I updated just this to math the other pattern.
Good catch @dariusk